### PR TITLE
Implement csr_matrix.eliminate_zeros

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented `csr_matrix.eliminate_zeros` using `csr2csr_compress`.

### CODE REVIEW

File: cupy/cuda/cusparse.pyx [hunk 1]
```diff
@@ -532,6 +554,54 @@ cpdef dcsr2dense(
...
+
+
+cpdef dnnz_compress(
+        size_t handle, int m, size_t descr,
+        size_t values, size_t rowPtr, size_t nnzPerRow,
+        size_t nnzTotal, double tol):
+    status = cusparseDnnz_compress(
+        <Handle>handle, m, <const MatDescr>descr,
+        <const double *>values, <const int *>rowPtr, <int *>nnzPerRow,
+        <int *>nnzTotal, tol)
```
Comment (okuta) [inline cupy/cuda/cusparse.pyx]: How about return `nnzTotal`?
Some CUDA runtime wrapper method return value.

Comment (unnonouno) [inline cupy/cuda/cusparse.pyx]: Actually this method has two return values `nnzPerRow` and `nnzTotal`. I think it is not natural to return only one of them.

Comment (okuta) [inline cupy/cuda/cusparse.pyx]: How about return tuple?

**Comment by unnonouno** [comment]
i fixed it

**Comment by okuta** [comment]
Sorry, I misunderstood.
`nnz_per_row` means array pointer on GPU. I think that it pointer is allocated by user, because user want to select memory type or reuse memory.
But, `nnz_total` means single value pointer on CPU.

**Comment by unnonouno** [comment]
How about adding an array argument to store result and return it like `out` argument in numpy?

**Comment by okuta** [comment]
I think it is OK.

**Comment by unnonouno** [comment]
`pyz` has a interface for `size_t`. I finally decided to use the first your idea.

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
I got build error on CUDA 7.0
```
cupy/cuda/cusparse.cpp:8380:258: error: 'cusparseSnnz_compress' was not declared in this scope
   __pyx_v_status = cusparseSnnz_compress(((cusparseHandle_t)__pyx_v_handle), __pyx_v_m, ((cusparseMatDescr_t const )__pyx_v_descr), ((float const *)__pyx_v_values), ((int const *)__pyx_v_rowPtr), ((int *)__pyx_v_nnzPerRow), (&__pyx_v_nnz_total), __pyx_v_tol);
                                                                                                                                                                                                                                                                  ^
cupy/cuda/cusparse.cpp: In function 'PyObject* __pyx_f_4cupy_4cuda_8cusparse_dnnz_compress(size_t, int, size_t, size_t, size_t, size_t, double, int)':
cupy/cuda/cusparse.cpp:8589:259: error: 'cusparseDnnz_compress' was not declared in this scope
   __pyx_v_status = cusparseDnnz_compress(((cusparseHandle_t)__pyx_v_handle), __pyx_v_m, ((cusparseMatDescr_t const )__pyx_v_descr), ((double const *)__pyx_v_values), ((int const *)__pyx_v_rowPtr), ((int *)__pyx_v_nnzPerRow), (&__pyx_v_nnz_total), __pyx_v_tol);
                                                                                                                                                                                                                                                                   ^
cupy/cuda/cusparse.cpp: In function 'PyObject* __pyx_f_4cupy_4cuda_8cusparse_scsr2csr_compress(size_t, int, int, size_t, size_t, size_t, size_t, int, size_t, size_t, size_t, size_t, float, int)':
cupy/cuda/cusparse.cpp:8797:377: error: 'cusparseScsr2csr_compress' was not declared in this scope
   __pyx_v_status = cusparseScsr2csr_compress(((cusparseHandle_t)__pyx_v_handle), __pyx_v_m, __pyx_v_n, ((cusparseMatDescr_t)__pyx_v_descrA), ((float const *)__pyx_v_inVal), ((int const *)__pyx_v_inColInd), ((int const *)__pyx_v_inRowPtr), __pyx_v_inNnz, ((int *)__pyx_v_nnzPerRow), ((float *)__pyx_v_outVal), ((int *)__pyx_v_outColInd), ((int *)__pyx_v_outRowPtr), __pyx_v_tol);
                                                                                                                                                                                                                                                                                                                                                                                         ^
cupy/cuda/cusparse.cpp: In function 'PyObject* __pyx_f_4cupy_4cuda_8cusparse_dcsr2csr_compress(size_t, int, int, size_t, size_t, size_t, size_t, int, size_t, size_t, size_t, size_t, float, int)':
cupy/cuda/cusparse.cpp:9059:379: error: 'cusparseDcsr2csr_compress' was not declared in this scope
   __pyx_v_status = cusparseDcsr2csr_compress(((cusparseHandle_t)__pyx_v_handle), __pyx_v_m, __pyx_v_n, ((cusparseMatDescr_t)__pyx_v_descrA), ((double const *)__pyx_v_inVal), ((int const *)__pyx_v_inColInd), ((int const *)__pyx_v_inRowPtr), __pyx_v_inNnz, ((int *)__pyx_v_nnzPerRow), ((double *)__pyx_v_outVal), ((int *)__pyx_v_outColInd), ((int *)__pyx_v_outRowPtr), __pyx_v_tol);
                                                                                                                                                                                                                                                                                                                                                                                           ^
error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
```

**Comment by unnonouno** [comment]
Hum, this function looks a new API.

**Comment by unnonouno** [comment]
This API was introduced from CUDA 8.0. I implemented `coo_matrix.eliminate_zeros` and fixed `csr_matrix` to use it when CUDA is old.

**Comment by okuta** [comment]
Please check `cupy-py35` test result.

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by okuta** [comment]
LGTM!

**Pull request merged**